### PR TITLE
Script for generating unit tests

### PR DESCRIPTION
**Pull request by ashetty1**
Issue #770

### CODE REVIEW

**Comment by ashetty1** [comment]
@cdrage Please see if this works.

File: script/test/cmd/generate_unit.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+KOMPOSE_ROOT=$(readlink -f $(dirname "${BASH_SOURCE}")/../../..)
+
+
+# Directory in which the output files have to be generated
+# Eg. script/test/fixtures/group-add/
+TEST_DIR=''
+
+# Location of the docker-compose file
+COMPOSE_FILE="${TEST_DIR}/docker-compose.yml"
```
Comment (cdrage) [inline script/test/cmd/generate_unit.sh]: most files use `yaml` I believe not `yml`

**Comment by cdrage** [comment]
This works, but perhaps lets make it interactive? Run `./script/make-test.sh` or something and it interacts with you and asks for the test directory?

File: script/test/cmd/generate_unit.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    exit 1;
+fi
+
+generate_k8s() {
+    kompose convert -f $COMPOSE_FILE -j -o $TEST_DIR/output-k8s.json
+    sed -i -e '/.*kompose.cmd.*:/ s/: .*/: "%CMD%"/' -e '/.*kompose.version.*:/ s/: .*/: "%VERSION%"/' ${TEST_DIR}/output-k8s.json
+}
+
+generate_os() {
+    kompose convert --provider=openshift -f $COMPOSE_FILE -j -o $TEST_DIR/output-os.json
```
Comment (cdrage) [inline script/test/cmd/generate_unit.sh]: please use `./kompose`

File: script/test/cmd/generate_unit.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+# One-line description for the test case
+TEST_DESCRIPTION=''
+
+if [ -z $TEST_DIR ] || [ -z $COMPOSE_FILE ] || [ -z $TEST_DESCRIPTION ]; then
+    echo "Please provide values for TEST_DIR, COMPOSE_FILE and TEST_DESCRIPTION in the script"
+    exit 1;
+fi
+
+generate_k8s() {
+    kompose convert -f $COMPOSE_FILE -j -o $TEST_DIR/output-k8s.json
```
Comment (cdrage) [inline script/test/cmd/generate_unit.sh]: please use `./kompose`

**Comment by ashetty1** [comment]
@cdrage @surajnarwade review please

**Comment by surajnarwade** [comment]
@ashetty1 , can you add it in Makefile and bit of documentation, I am confused how to test it

**Comment by ashetty1** [comment]
@surajnarwade Not sure if we should add it to Makefile. This is not a user-facing script.

**Comment by cdrage** [comment]
IMO, let's add it to makefile, it doesn't hurt adding more stuff to it.

**Comment by kadel** [comment]
> @surajnarwade Not sure if we should add it to Makefile. This is not a user-facing script.

It is not user-facing, but it is Kompose developer facing. This needs to be documented in [Development Guide](https://github.com/kubernetes/kompose/blob/master/docs/development.md)

**Comment by kadel** [comment]
> IMO, let's add it to makefile, it doesn't hurt adding more stuff to it. @ashetty1
>  👍 1

+1 for adding this to Makefile, otherwise it will be hard to find those scripts, it is much more convenient to run this from make

File: Makefile [hunk 4]
```diff
@@ -66,6 +66,11 @@ test-openshift:
 test-cmd:
 	./script/test/cmd/tests.sh
 
+# generate commandline tests
+.PHONY: gen-cmd
+gen-cmd:
```
Comment (cdrage) [inline Makefile]: may be better to do `generate-test-cmd` ? wdyt @kadel

**Comment by cdrage** [comment]
This LGTM. Let's merge and go from there.

**Pull request merged**